### ISSUE TITLE
learncpp.com: ads

### ISSUE THREAD

**Issue by KAGEYAM4** [uBlockOrigin/uAssets#18717]
### Prerequisites

- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/124740436/235392297-c4fc6290-73a6-4a70-96a9-3f6a7683f21d.png), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable external filter lists, or I have verified that the issue still occurs without enabling external filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL address of the web page

`https://www.learncpp.com/`

### Category

ads

### Description

Ads Placeholder still remains even though ads do not show up.

Additionaly if i click on those placeholder, the screen dims and mouse loading animation starts and then after few seconds it goes away.

### Other extensions used

![image](https://github.com/uBlockOrigin/uAssets/assets/75798544/61309f71-e340-483a-b8e6-41f0677bb70b)
![image](https://github.com/uBlockOrigin/uAssets/assets/75798544/540c5bb0-808b-4c0d-9c30-711ee5b1a64c)
![image](https://github.com/uBlockOrigin/uAssets/assets/75798544/22073d62-602e-4310-af39-c83be9397b1f)
![image](https://github.com/uBlockOrigin/uAssets/assets/75798544/c8731afc-126b-4ced-ab45-648bb3504df1)

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/75798544/9c63a390-12c1-4775-a40e-6c47da6cf209)

</details>

### Configuration

<details>

```yaml
uBlock Origin 1.50.0:
 since: 14d.47m
Firefox: 114
filterset (summary):
 network: 97226
 cosmetic: 84092
 scriptlet: 23175
 html: 1136
listset (total-discarded, last-updated):
 added:
  adguard-spyware-url: 1097-5, 4m
  fanboy-cookiemonster: 47643-156, never
  ublock-annoyances: 5259-7, 4m
 default:
  user-filters: 2-0, never
  ublock-filters: 34510-100, 4m
  ublock-badware: 7631-1, 4m
  ublock-privacy: 457-0, 4m
  ublock-unbreak: 2081-1, 4m
  easylist: 67585-739, 4m
  easyprivacy: 30615-710, 4m
  urlhaus-1: 6604-0, 4m
  plowe-0: 3739-1, 4m
  ublock-quick-fixes: 283-0, 4m
filterset (user): [array of 2 redacted]
trustedset:
 added: [array of 89 redacted]
hostRuleset:
 added: [array of 200 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings:
 filterAuthorMode: true
supportStats:
 allReadyAfter: 2924 ms (selfie)
 maxAssetCacheWait: 190 ms
popupPanel:
 blocked: 21
 network:
  learncpp.com: 12
  amazon-adsystem.com: 1
  doubleclick.net: 1
  ezodn.com: 2
  google-analytics.com: 1
  googlesyndication.com: 1
  pubmatic.com: 1
  quantserve.com: 1
  teads.tv: 1
 extended:
  ##[data-ez-name]
  ##[id^="google_ads_iframe"]
  ##div[id^="ezoic-pub-ad-"]
  ##span[id^="ezoic-pub-ad-placeholder-"]
  ##[id^="div-gpt-ad"]:not([style^="width: 1px; height: 1px; posit…
  ###ezmobfooter
  ##.ezmob-footer
  ##.ezoic-floating-bottom
  ##.ezo_ad
  ##.ezmob-footer-desktop
  ##.ezoic-ad
  ##.adtester-container
  #@#[id^="div-gpt-ad"]
  #@#.ad-unit
  #@#.banner-ads
  #@#.banner_ads
  #@#.ad-zone
  #@#.ad-space
```
</details>

**Comment by stephenhawk8054** [comment]
It's hidden correctly for me, screenshot: https://github.com/uBlockOrigin/uAssets/assets/66517106/f2654027-877c-4a98-bc08-4f2099cddc49

**Comment by krystian3w** [comment]
EL cookies addresed cookie bar by generic:

```
###ez-cookie-dialog-wrapper
```

same

AdGuard Cookie notices.

**Comment by okiehsch** [comment]
That doesn't help the user with a default setup.

**Comment by krystian3w** [comment]
So you menant all list should implement too:

`learncpp.com#@##ez-cookie-dialog-wrapper`

what function out of modal is broken without minimal/full accept?

**Comment by KAGEYAM4** [comment]
Both Webpages for issue mentioned above, now correctly works. One time it wasn't , now its working.

**Comment by krystian3w** [comment]
It is possible that they are testing a new way of injecting ads and, because of the failure, have revoked the reservation of ad space.

**Comment by KAGEYAM4** [comment]
Update - It's working in `container` but not otherwise

**Comment by okiehsch** [comment]
> So you meant all list should implemen

No. what I mean is we have to fix a broken site if it is broken with a default setup.

> what function out of modal is broken without minimal/full accept?

I can't click "Continue with Recommended Cookies" and without clicking that button the site will not function.

**Comment by krystian3w** [comment]
But I'm not asking why you're allowing a JS file, I'm asking what you have broken after using the filter:

`learncpp.com###ez-cookie-dialog-wrapper#ez-cookie-dialog-wrapper`

That allowing JS seems better.

**Comment by okiehsch** [comment]
>  I'm asking what you have broken after using the filter:
`learncpp.com###ez-cookie-dialog-wrapper#ez-cookie-dialog-wrapper`

The ability tointeract with  your GDPR settings.

**Comment by krystian3w** [comment]
Well, let you have this evasive answer (In this way, we can disprove the sense of the whole list of EL Cookies , that people will close themselves).

**Comment by okiehsch** [comment]
It's not evasive, we don't fix, i.e. hide cookie notices or GDPR modals for users with a default setup.
If I use your suggested filter that's exactly what I would be doing.

**Comment by okiehsch** [comment]
> It's hidden correctly for me, screenshot: https://github.com/uBlockOrigin/uAssets/assets/66517106/f2654027-877c-4a98-bc08-4f2099cddc49

Hm, it was the same on my end but now I can reproduce.
@KAGEYAM4 does it work for you if you add
`learncpp.com##.cf_monitor`
to your filter list?

**Comment by KAGEYAM4** [comment]
@okiehsch  now even i can't reproduce. It's working perfectly now. didn't even used `learncpp.com##.cf_monitor`.

**Comment by okiehsch** [comment]
Maybe it's a site issue. I will close for now, if the issue reappears just post it here.
